main_f4.c

```diff
@@ -377,7 +475,15 @@ main(void)
         }
 #endif
 
-#ifdef INTERFACE_USB
+	/* 
+	 * Check if the force-bootloader pins are strapped; if strapped,
+	 * don't try booting.
+	 */
+	if (board_test_force_pin()) {
```

Comment:
I this this ensures you can replace a good load of bad code. The previous code could have jumped to bad code that was loaded and had a good CRC. - @ksschwabe This is a very good find!